core/dvector.h

```diff
@@ -424,6 +425,17 @@ Error DVector<T>::resize(int p_size) {
 	return OK;
 }
 
+template<class T>
+void DVector<T>::invert() {
+	T temp;
+
+	for(int i=0;i<size()/2;i++) {
+		temp = get(i);
+		set(i, get(size()-i-1));
+		set(size()-i-1, temp);
+	}
```

Comment:
it would probably be better to use a locked vector for this, for performance

On Fri, May 13, 2016 at 11:23 AM, Bojidar Marinov <notifications@github.com>
wrote:

> In core/dvector.h
> <https://github.com/godotengine/godot/pull/4604#discussion_r63190716>:
>
> > @@ -424,6 +425,17 @@ Error DVector<T>::resize(int p_size) {
> >  	return OK;
> >  }
> >
> > +template<class T>
> > +void DVector<T>::invert() {
> > +	T temp;
> > +
> > +	for(int i=0;i<size()/2;i++) {
> > +		temp = get(i);
> > +		set(i, get(size()-i-1));
> > +		set(size()-i-1, temp);
> > +	}
>
> @kjohnson0451 <https://github.com/kjohnson0451> If it is before the loop,
> you will make size()/2 less calls to size(), since there are calls to it
> inside the loop as well.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/4604/files/0cf75451087014a1fcb32f3ce02099d0f6d6ecc9#r63190716>
>